src/core/MOM.F90

```diff
@@ -2547,16 +2555,18 @@ subroutine initialize_MOM(Time, Time_init, param_file, dirs, CS, restart_CSp, &
       deallocate(S_in)
     endif
     if (use_ice_shelf) &
-      deallocate(frac_shelf_in)
+      deallocate(frac_shelf_in,mass_shelf_in)
   else
     if (use_ice_shelf) then
       call initialize_ice_shelf(param_file, G, Time, ice_shelf_CSp, diag_ptr)
       allocate(CS%frac_shelf_h(isd:ied, jsd:jed), source=0.0)
-      call ice_shelf_query(ice_shelf_CSp,G,CS%frac_shelf_h)
+      allocate(CS%mass_shelf(isd:ied, jsd:jed), source=0.0)
+      call ice_shelf_query(ice_shelf_CSp,G,CS%frac_shelf_h, CS%mass_shelf)
+      CS%mass_shelf = CS%mass_shelf*US%kg_m3_to_R*US%m_to_Z
```

Comment:
The comments describing the arguments to ice_shelf_query indicate that mass_shelf is already being returned in units of [R Z ~> kg m-2], so reapplying the conversion factor on line 2565 would be a mistake.

Repeating a short run with non-zero values of R_RESCALE_POWER and Z_RESCALE_POWER (like 25 and -10) would be the surest way to verify whether it is the description of `ice_shelf_query()` or this unit conversion on line 2565 that needs to be corrected.

Reply:
I verified that the additional conversion factors caused model failure with rescaling applied - so agreed.